src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -551,10 +551,15 @@ struct
     in
     let root_ledger = Staged_ledger.ledger root in
     let soon_to_be_root_ledger = Staged_ledger.ledger soon_to_be_root in
-    let soon_to_be_root_merkle_root =
-      Ledger.merkle_root soon_to_be_root_ledger
+    let%bind soon_to_be_root_merkle_root =
+      Deferred.create (fun ivar ->
+          Ivar.fill ivar @@ Ledger.merkle_root soon_to_be_root_ledger )
+    in
+    let%map () =
+      Deferred.create (fun ivar ->
+          Ledger.commit soon_to_be_root_ledger ;
+          Ivar.fill ivar () )
     in
```

Comment:
the changes here are the same as writing `let%bind () = Async.Scheduler.yield in (* merkle root*) in let%map () = Async.Scheduler.yield() in (* ... *)` which I think is more clear.

Reply:
Ok, I can switch to that style.